src/app/project-admin/component/project-admin.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription, timer } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { NotificationService } from 'src/app/base/services/notification.service';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { RouteService } from 'src/app/base/services/route.service';
+import { UserManager } from 'src/app/util/user-manager';
+import { copyToClipboard } from 'src/app/util/helper-functions';
+import { ConfigManager } from 'src/app/base/services/config-service';
+
+@Component({
+  selector: 'project-admin-admin',
+  templateUrl: './project-admin.component.html',
+  styleUrls: ['./project-admin.component.scss']
+})
+
+export class ProjectAdminComponent implements OnInit {
+
+  projectName = new FormControl('');
+  project$: any;
+  projectQuery$: any;
+  project: any;
+  personalAccessTokens: any;
+  personalAccessTokensQuery$: any;
+  newToken: string;
+  tokenCopied: boolean = false;
+  tokenNameIsDuplicated: boolean;
+  subscriptions$: Subscription[] = [];
+
+  constructor(
+    private routeService: RouteService,
+    private activatedRoute: ActivatedRoute,
+    private projectApolloService: ProjectApolloService,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    this.checkUserIsAuthorized();
+    this.routeService.updateActivatedRoute(this.activatedRoute);
+
+    const projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(projectId);
+    this.project$.subscribe((project) => this.project = project);
```

Comment:
what is the project used for? i can only see the id (and HTML page) but nothing that really need the additional request -> i'd remove it and only store the projectId in a variable instead of a const

Reply:
Removed project because like you mentioned it is not used and therefore unnecessary call